Barotropic Streamfunction model agnostic

julia-neme opened the pull request:
barotropic streamfunction, depth integrated, in depth contours and de…nsity contours

navidcy commented:
I started reviewing this PR and noticed a few places where variable names were abbreviated without (apparent) reason?

E.g. why not use `experiment = 'pan_antarct...'` instead of `exp = ...`? It makes the code more readable and does not affect any performance.

navidcy commented:
> Appending a `.load()` to the end of this would force the computation to happen here, rather than at the plotting call.

I suggest we append `.load()` + a small remark why

julia-neme commented:
<p>Yes, you are right, but I couldn't find those distances in the diagnostics saved. However I did find <code>/g/data/ik11/inputs/mom6/panan-01/vcoord.nc</code> , but I wasn't sure if it was correct because it contains <code>st_ocean</code> and <code>st_ocean_levels</code>, which I thought was mom5 convention. Do you know where I could find this information?   </p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/228/discussion/#comment-1402821765'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

julia-neme commented:
<p>Yeah, I have a tendency to keep variable names short and simple... but I'll go through this and make longer, unique names now </p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/228/discussion/#comment-1402821773'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

julia-neme commented:
<p>Again even though its written right above?</p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/228/discussion/#comment-1402781632'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

navidcy commented:
perhaps leave it as is... it's a subjective matter anyways! :) :)

"short and simple" is a great principle

in my subjective opinion "exp" is short and ambiguous while "experiment" is short and simple

navidcy commented:
I really don't wanna impose my style here

I'm regretting I made the comment...

navidcy commented:
@julia-neme if you plan to work on this more then perhaps convert it to "Draft"?

navidcy commented:
<img width="845" alt="Screenshot 2023-02-06 at 12 06 01 pm" src="https://user-images.githubusercontent.com/7112768/216859239-51324ed1-d0ce-4604-b151-b22f9e20e7ad.png">

@julia-neme I see two files changes... I presume the one ending in `_model_agnostic` is what you meant to push?

julia-neme commented:
Yeah! I renamed the file from _MOM6.ipynb to _model_agnostic.ipynb and also moved it to the Documented_examples directory since it doesn't need to be in its own MOM6/ folder

navidcy commented:
@julia-neme I applaud the use of unicode `ψ` in code. Makes the code much more legible.

navidcy reviewed (changes requested):
@julia-neme great work!

I pushed a few minor phrasal changes.

I tried to run the notebook and it wasn't running though after cell 11. I think I fixed cell 12 but then I don't know.

Comment: After we plot the two stream functions in cells 10 and 11 aren't we done? You want to demonstrate something else further down? I suggest adding a sentence "What we did above is a simple way to compute the barotropic stream function... But let's delve into the nitty gritty now and do this and that." so that people know what's happening?

Also before you push perhaps try "Kernel > Restart and Run all cells"? That way you'll ensure that cells can be run in the order they appear.

navidcy commented:
I fixed a few cells by replacing 
```python
cc.querying.getvar(exp, 'uo', session,....
```
with 
```python
cc.querying.getvar(model_args['mom6']['expt'], 'uo', session,....
```
since `exp` was never defined. Not sure if I did what you were intending to do. Have a look. Thanks!

navidcy commented on DocumentedExamples/Barotropic_Streamfunction_model_agnostic.ipynb:
```diff
@@ -0,0 +1,1660 @@
+{
```
<p>this figure looks weird... </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/228/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

navidcy commented:
I think I made the notebook run until the end. But when you go into loading panantarct model output every calculation takes a bit. Perhaps add a remark on that? Something like "**Note**: next cell needs 2-3 min to evaluate"?

navidcy commented:
merge at your convenience! :)
